src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,202 @@
+
+            //if (previousProp.compareTo(propKey) > 0) {
+            if (String.CASE_INSENSITIVE_ORDER.compare(previousProp, propKey) > 0) {
+
+                final int lineNumber = getLineNumber(fileText, propKey);
+                log(lineNumber, MSG_KEY, propKey, previousProp);
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns the line number where the key is detected.
+     *
+     * @param fileText {@link FileText} object contains the lines to process
+     * @param keyName  key name to look for
+     * @return line number of first occurrence. If no key found in properties file, 0 is returned
+     */
+    private static int getLineNumber(FileText fileText, String keyName) {
+        final Pattern keyPattern = getKeyPattern(keyName);
+        int humanIndex = 0;
+
+        for (int index = 0; index < fileText.size(); index++) {
+            final String line = fileText.get(index);
+            if (keyPattern.matcher(line).matches()) {
+                // human line index
+                humanIndex = index + 1;
+                break;
+            }
+        }
+        // if no key is found return 0
+        return humanIndex;
+    }
+
+    /**
+     * Method returns regular expression pattern given key name.
+     *
+     * @param keyName
+     *            key name to look for
+     * @return regular expression pattern given key name
+     */
+    private static Pattern getKeyPattern(String keyName) {
+        final String keyPatternString = "^" + SPACE_PATTERN.matcher(keyName)
+                .replaceAll(Matcher.quoteReplacement("\\\\ ")) + "[\\s:=].*$";
+        return Pattern.compile(keyPatternString);
+    }
+
+    /**
+     * Private property implementation that keeps order of properties like in file.
+     *
+     * @noinspection ClassExtendsConcreteCollection, SerializableHasSerializationMethods
+     */
+    private static class SequencedProperties extends Properties {
+
+        private static final long serialVersionUID = 1L;
+
+        /**
+         * Holding the keys in the same order than in the file.
+         */
+        private final List<Object> keyList = new ArrayList<>();
+
+        /**
+         * Returns a copy of the keys.
+         */
+        @Override
+        public synchronized Enumeration<Object> keys() {
+            return Collections.enumeration(keyList);
+        }
+
+        /**
+         * Puts the value into list by its key.
+         * @noinspection UseOfPropertiesAsHashtable
+         *
+         * @param key the hashtable key
+         * @param value the value
+         * @return the previous value of the specified key in this hashtable,
+         *      or null if it did not have one
+         * @Throws NullPointerException - if the key or value is null
+         */
+        @Override
+        public synchronized Object put(Object key, Object value) {
+            if (!(keyList.contains(key))) {
```

Comment:
> `+OrderedPropertiesCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (!(keyList.contains(key))) {</span></pre></td></tr>`

One of your tests must have a property file with a key that appears twice in the file. Since you remove duplicate keys, you must show that the 2nd instance of the key won't produce a violation if it is also out of order like the 1st instance.

Reply:
I added in a property file an double key. I removed this if, because it works also if the list is non unique.